fast64_internal/oot/oot_actor.py

```diff
@@ -1,47 +1,424 @@
 
+	def execute(self, context):
+		for listNode in root:
+			if listNode.tag == 'List' and listNode.get('Name') == 'Elf_Msg Message ID':
+				for elem in listNode:
+					if elem.get('Key') == self.naviMsgID:
+						bpy.data.objects[self.objName].fast64.oot.actor.naviMsgID = elem.get('Value')
 		bpy.context.region.tag_redraw()
-		self.report({'INFO'}, "Selected: " + self.actorID)
+		self.report({'INFO'}, "Selected: " + self.naviMsgID)
 		return {'FINISHED'}
 
 	def invoke(self, context, event):
 		context.window_manager.invoke_search_popup(self)
 		return {'RUNNING_MODAL'}
 
+class OOTActorProperties(bpy.types.PropertyGroup):
+	# Each prop has its 'custom' variant because of the get and set functions
+	# Actors/Entrance Actor
+	actorID : bpy.props.StringProperty(name='Actor ID', default='ACTOR_PLAYER')
+	actorKey : bpy.props.EnumProperty(name='Actor Key', items=ootEnumActorID)
+	actorParam : bpy.props.StringProperty(name = 'Actor Parameter', default = '0x0000',
+		get=lambda self: getValues(self, self.actorKey, 'actorParam', 'Params', None),
+		set=lambda self, value: setValues(self, value, 'Params', 'actorID'))
+	actorParamCustom : bpy.props.StringProperty(name = 'Actor Parameter', default = '0x0000')
+
+	# Rotations
+	rotOverride : bpy.props.BoolProperty(name = 'Rot Override', default = False)
+	rotOverrideX : bpy.props.StringProperty(name = 'Rot X', default = '0x0',
+		get=lambda self: getValues(self, self.actorKey, 'rotOverrideX', 'XRot', None),
+		set=lambda self, value: setValues(self, value, 'XRot', 'actorID'))
+	rotOverrideY : bpy.props.StringProperty(name = 'Rot Y', default = '0x0',
+		get=lambda self: getValues(self, self.actorKey, 'rotOverrideY', 'YRot', None),
+		set=lambda self, value: setValues(self, value, 'YRot', 'actorID'))
+	rotOverrideZ : bpy.props.StringProperty(name = 'Rot Z', default = '0x0',
+		get=lambda self: getValues(self, self.actorKey, 'rotOverrideZ', 'ZRot', None),
+		set=lambda self, value: setValues(self, value, 'ZRot', 'actorID'))
+	rotOverrideXCustom : bpy.props.StringProperty(name = 'Rot X', default = '0x0')
+	rotOverrideYCustom : bpy.props.StringProperty(name = 'Rot Y', default = '0x0')
+	rotOverrideZCustom : bpy.props.StringProperty(name = 'Rot Z', default = '0x0')
+
+	# Transition Actors
+	transActorID : bpy.props.StringProperty(name='Transition Actor ID', default='ACTOR_EN_DOOR')
+	transActorKey : bpy.props.EnumProperty(name='Transition Actor ID', items=ootEnumTransitionActorID)
+	transActorParam : bpy.props.StringProperty(name = 'Actor Parameter', default = '0x0000',
+		get=lambda self: getValues(self, self.transActorKey, 'actorParam', 'Params', 'transActorParam'),
+		set=lambda self, value: setValues(self, value, 'Params', 'transActorID'))
+	transActorParamCustom : bpy.props.StringProperty(name = 'Actor Parameter', default = '0x0000')
+
+	# Other
+	itemChest : bpy.props.StringProperty(name='Chest Content', default = '0x48')
+	naviMsgID : bpy.props.StringProperty(name='Navi Message ID', default = '0x00')
```

Comment:
I think the default being an integer is a mistake, when you use the search prop it sets the key

and using enums would be painful because of how large these are, it's faster to just type "heart" or "wake up!" (especially for navi messages)